src/lib.rs

```diff
@@ -14,6 +14,10 @@ pub trait StalinFind<T> {
     if self.is_empty() {
       None
     } else {
+      self.sort() // so that the "binarity" of the search is finally useful
```

Comment:
no, this increases complexity a lot, I think it's meaningless to do, you can use alternative find algos if array isn't sorted or not fully sorted.

Reply:
> you may have luck that this algo will work fast as well

sure. as one might have with the bogglesort